File: data/java/tests/test_itertools_product.java
package com.example;

import org.junit.jupiter.api.BeforeEach;
import org.junit.jupiter.api.Test;
import static org.junit.jupiter.api.Assertions.*;
import java.util.Arrays;
import java.util.stream.Collectors;

public class test_itertools_product {
    private function_itertools_product processor;

    @BeforeEach
    void setUp() {
        processor = new function_itertools_product();
    }

    @Test
    void testCartesianProductOneElementInThreeLists() {
        String input = "1;2,3;4,5";
        String expectedResult = "124;125;134;135";
        String result = processor.test_product(input);
        assertEquals(
                Arrays.stream(expectedResult.split(";")).sorted().collect(Collectors.joining(";")),
                Arrays.stream(result.split(";")).sorted().collect(Collectors.joining(";"))
        );
    }

    @Test
    void testCartesianProductNegativeNumbers() {
        String input = "-1,0;-2,2";
        String expectedResult = "-1-2;-12;0-2;02";
        String result = processor.test_product(input);
        assertEquals(
                Arrays.stream(expectedResult.split(";")).sorted().collect(Collectors.joining(";")),
                Arrays.stream(result.split(";")).sorted().collect(Collectors.joining(";"))
        );
    }

    @Test
    void testCartesianProductNoneInput() {
        assertThrows(IllegalArgumentException.class, () -> processor.test_product(null));
    }

    @Test
    void testCartesianProductNonIterableInput() {
        // 修改这里：将整数123转换为字符串"123"
        assertThrows(IllegalArgumentException.class, () -> processor.test_product("123"));
    }

    @Test
    void testCartesianProductElementsNotIterable() {
        assertThrows(IllegalArgumentException.class, () -> processor.test_product("1,2;;4,5"));
    }
}

File: manual/data/python/tests/test_itertools_product.py
import unittest
from function_itertools_product import CartesianProductProcessor


class TestCartesianProduct(unittest.TestCase):

    def setUp(self):
        self.processor = CartesianProductProcessor()

    def test_cartesian_product_one_element_in_three_lists(self):
        input_string = "1;2,3;4,5"
        expected_result = "124;125;134;135"
        result = self.processor.test_product(input_string)
        self.assertEqual(sorted(result.split(';')), sorted(expected_result.split(';')))

    def test_cartesian_product_negative_numbers(self):
        input_string = "-1,0;-2,2"
        expected_result = "-1-2;-12;0-2;02"
        result = self.processor.test_product(input_string)
        self.assertEqual(sorted(result.split(';')), sorted(expected_result.split(';')))

    def test_cartesian_product_none_input(self):
        with self.assertRaises(ValueError):
            self.processor.test_product(None)

    def test_cartesian_product_non_iterable_input(self):
        with self.assertRaises(ValueError):
            self.processor.test_product(123)

    def test_cartesian_product_elements_not_iterable(self):
        with self.assertRaises(ValueError):
            self.processor.test_product("1,2;;4,5")